src/core/server/saved_objects/export/saved_objects_exporter.ts

```diff
@@ -99,6 +108,12 @@ export class SavedObjectsExporter {
     let exportedObjects: Array<SavedObject<unknown>>;
     let missingReferences: SavedObjectsExportResultDetails['missingReferences'] = [];
 
+    savedObjects = await applyExportHooks({
```

Comment:
`applyExportHooks` will change the sorting order. The new sorting order will be consistent, but it would be nice if we could keep the sorting order the same as previous versions because of https://github.com/elastic/kibana/issues/29747.

Reply:
Maybe you could also add a comment to the code, because it's not so obvious from the existing code why we did this sort in the first place.

Reply:
Would be great, however as the hooks mutates subsets of the exported objects, I'm not really sure how to do this without keeping a deepCopy of the object and manually comparing the two list to see changes and additions to try to preserve the order. Do you see any 'easy' way to preserve the order?

Reply:
I just realized that types using the export hooks would not have ever been exported before, so for version control it won't really matter (it's not impossible for `hidden: false` objects to also adopt export hooks, but probably unlikely). However, plugins implementing an exportHook might not know/forget to use a stable sort order. So I think it's best if we:
 - keep the current sort in `fetchByTypes`
 - also sort all objects returned by an export hook

Reply:
> I just realized that types using the export hooks would not have ever been exported before, so for version control it won't really matter

The hooks handling logic still changes the position of types not having registered hooks (see https://github.com/elastic/kibana/pull/87807/files#diff-d6563061094fb926296a7960b2c76ec062d04e63e22646fb48975ab03dde6fe5). So we still need to find a way to reorder all the objects if we want to be compliant with https://github.com/elastic/kibana/issues/29747